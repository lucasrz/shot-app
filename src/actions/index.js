import { GET_SHOTS_SAGA, SET_SHOTS, SET_SHOT, GET_SHOT_SAGA, CLEAR_STORE } from '../constants';

export function setShots(shots) {
  return {
    type: SET_SHOTS,
    shots
  };
}

export function clearStore() {
  return {
    type: CLEAR_STORE
  };
}

export function setShot(shot) {
  return {
    type: SET_SHOT,
    shot
  };
}

export function getShotSaga(id) {
  return {
    type: GET_SHOT_SAGA,
    id
  };
}

//Sagas
export function getShotsSaga() {
  return {
    type: GET_SHOTS_SAGA
  };
}
