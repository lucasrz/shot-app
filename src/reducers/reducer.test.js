import reducer from './index';
import { setShots } from '../actions';
import { setShot } from '../actions';
import { clearStore  } from '../actions';
import workerGetShotsSaga  from '../sagas/watchers/getShots';
import  workerGetShotSaga from '../sagas/watchers/getShots';

describe('Reducers', () => {
  it('Set shots', () => {
    const initialState = {};
    const next = reducer(initialState, setShots(['1','2','3']))
    expect(next).toMatchSnapshot()
  });

  it('Set shot', () => {
    const initialState = {};
    const next = reducer(initialState, setShot({}))
    expect(next).toMatchSnapshot()
  });

  it('Clear store', () => {
    const initialState = {};
    const next = reducer(initialState, clearStore());
    expect(next).toMatchSnapshot();
  });

  it('Get shots saga', () => {
    const initialState = {};
    const next = workerGetShotsSaga();
    expect(next).toMatchSnapshot();
  });

  it('Get shot saga', () => {
    const next = workerGetShotSaga({id: '6680212'});
    expect(next).toMatchSnapshot();
  });
});