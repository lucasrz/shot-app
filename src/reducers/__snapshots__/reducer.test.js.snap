// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Clear store 1`] = `
Object {
  "shotsReducer": Object {
    "undefined": undefined,
  },
}
`;

exports[`Reducers Get shot saga 1`] = `Object {}`;

exports[`Reducers Get shots saga 1`] = `Object {}`;

exports[`Reducers Set shot 1`] = `
Object {
  "shotsReducer": Object {
    "shot": Object {},
    "shots": Array [],
  },
}
`;

exports[`Reducers Set shots 1`] = `
Object {
  "shotsReducer": Object {
    "shots": Array [
      "1",
      "2",
      "3",
    ],
  },
}
`;
